File: ENC221-0087-2024 - ICS2276 Week 5 Practical.cpp
#include <iostream>
using namespace std;
struct fraction{
    int numerator;
    int denominator;

    //Decimal compuation function
    double value() const {
        return static_cast <double>(numerator)/denominator;
    }
};
// Function to find largest function using the return by reference
fraction& largestFraction(fraction &f1, fraction &f2, fraction &f3 ){
 fraction *largest = &f3; //assuming f3 is initially the largest

 if (f2.value()> largest->value()){
    largest = &f2;
 }
if (f1.value() > largest->value()){
    largest = &f1;
}
return  *largest; //return the reference t the largest fraction
}
//pass by reference ensures that fractions are passed by reference and thus no copies are made
int main(){
    fraction f1 = {5, 7};
    fraction f2 = {6, 8};
    fraction f3 = {1, 5};
    
    fraction &largest = largestFraction(f1,f2,f3);
    cout<< "Largest fraction is: "
    << largest.numerator <<"/"<< largest.denominator << endl;

    return 0;
}

File: ENC221-0087-2024 - ICS2276 Week 5 Practical.py
class Fraction:
 def __init__(self, numerator, denominator) :
    self.numerator = numerator
    self.denominator = denominator

 def value(self):
    return self.numerator / self. denominator

 def __str__(self):
    return f"{self.numerator}/{self.denominator}"

def largest_fraction(f1,f2, f3):
   largest = f1
   if f2.value() > largest. value() :
      largest = f2
   if f3.value() > largest. value():
      largest = f3

   return largest
   
#usage
f1 = Fraction(3, 15)
f2 = Fraction(7, 35)
f3 = Fraction(9,45)

largest = largest_fraction(f1, f2, f3)
print("Largest fraction is:", largest)
